File: Battleship.Client/src/services/SignalR/chat-hub.ts
import { Hub } from './hub'

export class ChatHub extends Hub{
    constructor(){
        super("chathub")
    }

    registerOnServerEvents(): void {    
        this.hubConnection.on('receiveMessage', (user: string, message: string) => {
            console.log(user + ': '+message);
      });
    }

    public sendMessage(user: string, message: string){
        this.hubConnection.send("sendMessage", user, message)
        .then(() => console.log("message sent"));
    }
}

File: BattleQuiz.Server/Hubs/ChatHub.cs
using System;
using System.Threading.Tasks;
using Microsoft.AspNetCore.SignalR;

namespace BattleQuiz.Server.Hubs
{
    public class ChatHub : Hub<IChatClient>
    {
        public Task SendMessage(string user, string message)
        {
            return Clients.All.ReceiveMessage(user, message);
        }

        public override Task OnConnectedAsync()
        {
            Console.WriteLine($"New connection {Context.ConnectionId} {Context.UserIdentifier}");
            return base.OnConnectedAsync();
        }
    }
}